prow/cmd/tot/main.go

```diff
@@ -255,7 +255,12 @@ func (f fallbackHandler) getURL(jobName string) string {
 		logrus.Errorf("requested job is unknown to prow: %s", jobName)
 		return ""
 	}
-	return fmt.Sprintf("%s/%s", strings.TrimSuffix(f.bucket, "/"), gcs.LatestBuildForSpec(spec))
+	paths := gcs.LatestBuildForSpec(spec, nil)
+	if len(paths) != 1 {
```

Comment:
You should make this `< 1`, and `at least one GCS path`. Presubmits using the first returned value is correct.

Reply:
`tot` should not use the PR-specific path. What if the request for a build number comes from a batch job?

Reply:
It's not PR-specific. The first return value for presubmits and batch jobs is `path.Join("pr-logs", "directory", spec.Job, "latest-build.txt")`.

Reply:
Correct, the second path returned though is PR-specific, that's why I pass a `nil` repobuilder in `LatestBuildForSpec` to force it to return a single path that is agnostic to the repobuilder.

Reply:
Ah, missed that subtlety.